standard/expressions.md

```diff
@@ -36,6 +36,142 @@ Most of the constructs that involve an expression ultimately require the express
+
+A pattern may have one of the following forms:
+
+```ANTLR
+pattern
+    : declaration_pattern
+    | constant_pattern
+    | var_pattern
+    ;
+```
+
+A *declaration_pattern* and a *var_pattern* can result in the declaration of a local variable. The scope of such a variable is as follows:
+- If the pattern is a case label ([§13.8.3](statements.md#1383-the-switch-statement)), then the scope of the variable is the associated *case block*.
+- Otherwise, the variable is part of the pattern that is the right-hand operand of the `is` operator ([§12.11.11](expressions.md#121111-the-is-operator)), and the variable’s scope is based on the construct immediately enclosing the `is` expression containing the pattern, as follows:
+  -  If the expression is in an expression-bodied lambda, the variable's scope is the body of the lambda.
+  -  If the expression is in an expression-bodied method or property, the variable's scope is the body of the method or property.
+  -  If the expression is in a `when` clause of a `catch` clause, the variable's scope is that `catch` clause.
+  -  If the expression is in an *iteration_statement*, the variable's scope is just that statement.
+  -  If the expression is in a *constructor_initializer*, the variable's scope is the part of that *constructor_initializer* following the expression, and the body of the constructor.
+  -  If the expression is in a *variable_initializer* of a field, the variable's scope is the part of that *variable_initializer* following the expression.
+  -  If the expression is in a *query_expression* that is translated into the body of a lambda, the variable's scope is that *query_expression*.
+  -  Otherwise if the expression is in some other statement form, the variable's scope is the scope containing the statement.
+
+For the purpose of determining the scope, an *embedded_statement* is considered to be in its own scope.
+
+#### §declaration-pattern-new-clause Declaration pattern
+
+A *declaration_pattern* is used to test that a value has a given type and if the test succeeds, to cast that value to that type. 
+
+```antlr
+declaration_pattern
+    : type simple_designation
+    ;
+simple_designation
+    : single_variable_designation
+    ;
+single_variable_designation
+    : identifier
+    ;
+```
+
+The runtime type of the value is tested against the *type* in the pattern. If it is of that runtime type (or some subtype), the result of the `is` operator is `true`. A pattern input value with value `null` never tests true for this pattern.
```

Comment:
Rather than saying "the result of the `is` operator is true" ... because the pattern might not be part of an is operator ... we should say that the pattern *matches*.  And for the is-pattern operator, we should say that it returns true if and only if the pattern *matches* on the input.

Reply:
Or maybe *suceeds*

Reply:
@gafter – Go with your first thought. I think “pattern *matches*” is the more widely used phrasing and we should stick with it. Though you might also see “if the pattern matching succeeds/is successful”…  ;-)